src/Application/User/Command/ProConnect/CreateProConnectUserCommandHandler.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Application\User\Command\ProConnect;
+
+use App\Application\ApiOrganizationFetcherInterface;
+use App\Application\DateUtilsInterface;
+use App\Application\IdFactoryInterface;
+use App\Domain\User\Enum\OrganizationRolesEnum;
+use App\Domain\User\Enum\UserRolesEnum;
+use App\Domain\User\Exception\OrganizationNotFoundException;
+use App\Domain\User\Organization;
+use App\Domain\User\OrganizationUser;
+use App\Domain\User\ProConnectUser;
+use App\Domain\User\Repository\OrganizationRepositoryInterface;
+use App\Domain\User\Repository\OrganizationUserRepositoryInterface;
+use App\Domain\User\Repository\ProConnectUserRepositoryInterface;
+use App\Domain\User\Repository\UserRepositoryInterface;
+use App\Domain\User\User;
+
+final class CreateProConnectUserCommandHandler
+{
+    public function __construct(
+        private IdFactoryInterface $idFactory,
+        private UserRepositoryInterface $userRepository,
+        private ProConnectUserRepositoryInterface $proConnectUserRepository,
+        private OrganizationUserRepositoryInterface $organizationUserRepository,
+        private OrganizationRepositoryInterface $organizationRepository,
+        private DateUtilsInterface $dateUtils,
+        private ApiOrganizationFetcherInterface $organizationFetcher,
+    ) {
+    }
+
+    public function __invoke(CreateProConnectUserCommand $command): User
+    {
+        ['given_name' => $givenName, 'family_name' => $familyName, 'organization_siret' => $siret] = $command->userInfo;
+
+        $user = $this->userRepository->findOneByEmail($command->email);
+        if ($user instanceof User) {
+            return $user;
+        }
+
+        $organization = $this->organizationRepository->findOneBySiret($siret);
+        $organizationRole = OrganizationRolesEnum::ROLE_ORGA_CONTRIBUTOR->value; // Default organization role
```

Comment:
Définir le rôle par défaut sur le OrganizationRolesEnum ? Pour ne pas dupliquer entre ici et le RegisterCommandHandler
```suggestion
        $organizationRole = OrganizationRolesEnum::ROLE_DEFAULT_FOR_CREATE_USER->value;
```